pkg/tracing/logger.go

```diff
@@ -0,0 +1,47 @@
+package tracing
+
+import (
+	"github.com/go-logr/logr"
+	ot "github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+)
+
+type tracingLogger struct {
+	logr.Logger
+	ot.Span
+}
+
+func (t tracingLogger) Enabled() bool {
+	return t.Logger.Enabled()
+}
+
+func (t tracingLogger) Info(msg string, keysAndValues ...interface{}) {
+	t.Logger.Info(msg, keysAndValues...)
+	t.Span.LogKV(append([]interface{}{"info", msg}, keysAndValues...)...)
+}
+
+func (t tracingLogger) Error(err error, msg string, keysAndValues ...interface{}) {
+	t.Logger.Error(err, msg, keysAndValues...)
+	t.Span.LogKV(append([]interface{}{"error", err, "message", msg}, keysAndValues...)...)
```

Comment:
is there a way to avoid needing an extra allocation here?

Reply:
also, is there a semantic difference between `WithTags` and the key-value pairs specified here?

Reply:
That `LogKV()` call is through an interface so the memory will escape; it's unlikely to be possible to avoid an allocation. This particular line is only hit under error conditions so probably not performance-sensitive, but there are similar ones.

Tags apply to the whole span so it wouldn't be right to put details from one error as a tag, however the next line sets an `error` tag on the span so you can pick it out as one that had at least one error.